reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -342,9 +513,18 @@ private void populateAdvancedArtifactReportFields(AdvancedArtifactReport advance
         }
     }
 
-    private Set<String> getWhitelistedVersions(String groupId, String artifactId) {
-        return productVersionService.getProductVersionsWithArtifactsByGA(groupId, artifactId)
-                .stream()
+    private Set<String> getWhitelistedVersions(String groupId, String artifactId, Set<Long> validProductVersionIds) {
+        
+        List<ProductVersionArtifactRelationship> prodVerRels = productVersionService.getProductVersionsWithArtifactsByGA(groupId, artifactId);
+        
+        // If values are present restrict results based on id
+        if(validProductVersionIds != null && !validProductVersionIds.isEmpty()){
```

Comment:
If user enters invalid product name or nonexistent productVersion id, this filtration doesn't happen so user gets info from all whitelists. Is it correct?